sdk/src/dex/orca_legacy.ts

```diff
@@ -0,0 +1,276 @@
+import {
+    Cluster,
+    Keypair,
+    PublicKey,
+    SystemProgram,
+    Transaction,
+    TransactionInstruction,
+} from "@solana/web3.js";
+import { TOKEN_PROGRAM_ID, Token } from "@solana/spl-token";
+import * as BufferLayout from "@solana/buffer-layout";
+import * as anchor from "@project-serum/anchor";
+import { OrcaPoolConfig } from "@orca-so/sdk";
+import { orcaPoolConfigs } from "@orca-so/sdk/dist/constants/pools";
+import { OrcaPoolParams } from "@orca-so/sdk/dist/model/orca/pool/pool-types";
+
+export interface OrcaLegacyAccounts {
+    marketId: number;
+    programId: PublicKey;
+    swapProgram: PublicKey;
+    swapAuthority: PublicKey;
+    poolTokenMint: PublicKey;
+    feeAccount: PublicKey;
+    tokenAccountA: PublicKey;
+    tokenAccountB: PublicKey;
+    vaultOrcaLegacyAccount?: PublicKey;
+}
+
+export class OrcaLegacySwap {
+    private constructor(public accounts: OrcaLegacyAccounts) {}
+
+    static load(
+        tokenA: PublicKey,
+        tokenB: PublicKey,
+        cluster: Cluster
+    ): OrcaLegacySwap {
+        const tokenPairSig = tokenA.toString() + tokenB.toString();
+        let tokenPairToOrcaLegacyPool;
+
+        if (cluster == "devnet") {
+            // TODO mock orca pool
+        } else if (cluster == "mainnet-beta") {
+            // Load orca pool parameters from the sdk for look-up.
+            // Because the sdk doesn't support look-up using token mint pubkeys.
+            tokenPairToOrcaLegacyPool = Object.fromEntries(
+                Object.values(OrcaPoolConfig).map((v) => {
+                    const params = orcaPoolConfigs[v];
+                    const tokens = Object.keys(params.tokens);
+                    return [
+                        tokens[0].toString() + tokens[1].toString(),
+                        params,
+                    ];
+                })
+            );
+        } else {
+            throw new Error("Cluster ${cluster} not supported");
+        }
+
+        const params: OrcaPoolParams = tokenPairToOrcaLegacyPool[tokenPairSig];
+        if (params == undefined) {
+            throw new Error("Token pair not supported");
+        }
+
+        // TODO marketId to trading pair mapping
+        const marketId = 0;
+
+        const accounts = {
+            marketId: marketId,
+            programId: DEVNET_ORCA_TOKEN_SWAP_ID,
+            swapProgram: params.address,
+            swapAuthority: params.authority,
+            poolTokenMint: params.poolTokenMint,
+            feeAccount: params.feeAccount,
+            tokenAccountA: tokenA,
+            tokenAccountB: tokenB,
+        };
+        return new OrcaLegacySwap(accounts);
+    }
+
+    // This is used ONLY to create a mock orca swap for testing
+    static async initialize(
```

Comment:
optional: idk if we expect anyone to use the SDK other than us, but we might want to start renaming stuff like this to initializeMockSwap or something

Reply:
You are right, this will probably not be used by anyone else. But I probably leave it for another time since it's not related to this feature.